src/coreclr/hosts/unixcoreconsole/CMakeLists.txt

```diff
@@ -18,12 +18,13 @@ if(NOT CMAKE_SYSTEM_NAME STREQUAL FreeBSD)
     )
 endif(NOT CMAKE_SYSTEM_NAME STREQUAL FreeBSD)
 
-# FreeBSD requires pthread to be loaded by the executable process
-if(CMAKE_SYSTEM_NAME STREQUAL FreeBSD)
-    target_link_libraries(coreconsole
-        pthread
-    )
-endif(CMAKE_SYSTEM_NAME STREQUAL FreeBSD)
+# Loading pthread into the process is necessary to make
+# external libraries used by PInvoke and depending on
```

Comment:
This problem is not limited to PInvoke - I would make the comment more clear:

libc turns locks into no-ops if pthread was not loaded into process yet. Loading
pthread by the process executable ensures that all locks are initialized properly.